admin/load.php

```diff
@@ -636,3 +639,40 @@ function perflab_plugin_admin_notices() {
 	}
 }
 add_action( 'admin_notices', 'perflab_plugin_admin_notices' );
+
+/**
+ * Callback function that print plugin activation script.
+ *
+ * @since n.e.x.t
+ */
+function perflab_print_plugin_activation_script() {
+	ob_start();
+	?>
+	( function( $ ) {
+		$( document ).ajaxComplete( function( event, xhr, settings ) {
+			// Check if this is the 'install-plugin' request.
+			if ( settings.data && typeof settings.data === 'string' && settings.data.includes( 'action=install-plugin' ) ) {
+				var target_element = $( event.target.activeElement );
+				// Bail early if the element is not a[href].
+				if ( ! target_element.is( 'a[href]' ) ) {
+					return;
+				}
+				/*
+				 * WordPress core uses a 1s timeout for updating the activation link,
+				 * so we set a 1.5 timeout here to ensure our changes get updated after
+				 * the core changes have taken place.
+				 */
+				setTimeout( function() {
+					var plugin_url = target_element.attr( 'href' );
+					var nonce = target_element.attr( 'data-plugin-activation-nonce' );
+					var url = new URL( plugin_url );
+					url.searchParams.set( 'action', 'perflab_activate_plugin' );
+					url.searchParams.set( '_wpnonce', nonce );
+					target_element.attr( 'href', url.href );
+				}, 1500 );
+			}
+		} );
+	} )( jQuery );
+	<?php
+	wp_print_inline_script_tag( ob_get_clean() );
```

Comment:
This will retain syntax highlighting in most IDEs. See https://core.trac.wordpress.org/ticket/59444

```suggestion
	$js = <<<JS
	?>
	( function( $ ) {
		$( document ).ajaxComplete( function( event, xhr, settings ) {
			// Check if this is the 'install-plugin' request.
			if ( settings.data && typeof settings.data === 'string' && settings.data.includes( 'action=install-plugin' ) ) {
				var target_element = $( event.target.activeElement );
				// Bail early if the element is not a[href].
				if ( ! target_element.is( 'a[href]' ) ) {
					return;
				}
				/*
				 * WordPress core uses a 1s timeout for updating the activation link,
				 * so we set a 1.5 timeout here to ensure our changes get updated after
				 * the core changes have taken place.
				 */
				setTimeout( function() {
					var plugin_url = target_element.attr( 'href' );
					var nonce = target_element.attr( 'data-plugin-activation-nonce' );
					var url = new URL( plugin_url );
					url.searchParams.set( 'action', 'perflab_activate_plugin' );
					url.searchParams.set( '_wpnonce', nonce );
					target_element.attr( 'href', url.href );
				}, 1500 );
			}
		} );
	} )( jQuery );
JS;
	wp_print_inline_script_tag( $js );
```

Reply:
Thanks for sharing.